### PR TITLE
feat: upgrade to openresty-1.25.3.1

### PR DESCRIPTION
**Pull request by zll600**
related PRs
- https://github.com/api7/ngx_multi_upstream_module/pull/12
- https://github.com/api7/ngx_multi_upstream_module/pull/13
- https://github.com/api7/grpc-client-nginx-module/pull/73
- https://github.com/api7/wasm-nginx-module/pull/162
- https://github.com/api7/apisix-nginx-module/pull/83

### CODE REVIEW

File: build-apisix-runtime.sh [hunk 1]
```diff
@@ -91,7 +91,7 @@ fi
 if [ "$repo" == ngx_multi_upstream_module ]; then
     cp -r "$prev_workdir" ./ngx_multi_upstream_module-${ngx_multi_upstream_module_ver}
 else
-    git clone --depth=1 -b $ngx_multi_upstream_module_ver \
+    git clone --depth=1 \
```
Comment (zll600) [inline build-apisix-runtime.sh:94]: only for enable ci test.

File: build-apisix-runtime.sh [hunk 2]
```diff
@@ -107,8 +107,8 @@ fi
 if [ "$repo" == apisix-nginx-module ]; then
     cp -r "$prev_workdir" ./apisix-nginx-module-${apisix_nginx_module_ver}
 else
-    git clone --depth=1 -b $apisix_nginx_module_ver \
-        https://github.com/api7/apisix-nginx-module.git \
+    git clone --depth=1 -b upgrade_openresty_1.25.3.1 \
+        https://github.com/zll600/apisix-nginx-module.git \
```
Comment (zll600) [inline build-apisix-runtime.sh:111]: only for enable ci test.

File: build-apisix-runtime.sh [hunk 3]
```diff
@@ -91,8 +91,8 @@ fi
 if [ "$repo" == ngx_multi_upstream_module ]; then
     cp -r "$prev_workdir" ./ngx_multi_upstream_module-${ngx_multi_upstream_module_ver}
 else
-    git clone --depth=1 -b $ngx_multi_upstream_module_ver \
-        https://github.com/api7/ngx_multi_upstream_module.git \
+    git clone --depth=1 -b upgrade_openresty-1.25.3 \
+        https://github.com/zll600/ngx_multi_upstream_module.git \
```
Comment (monkeyDluffy6017) [inline build-apisix-runtime.sh:95]: please revert this

File: build-apisix-base.sh [hunk 4]
```diff
@@ -82,8 +82,8 @@ fi
 if [ "$repo" == grpc-client-nginx-module ]; then
     cp -r "$prev_workdir" ./grpc-client-nginx-module-${grpc_client_nginx_module_ver}
 else
-    git clone --depth=1 -b $grpc_client_nginx_module_ver \
-        https://github.com/api7/grpc-client-nginx-module \
+    git clone --depth=1 -b fix_header_files \
+        https://github.com/zll600/grpc-client-nginx-module \
```
Comment (monkeyDluffy6017) [inline build-apisix-base.sh:86]: please revert this

File: build-apisix-base.sh [hunk 5]
```diff
@@ -58,16 +58,16 @@ fi
...
         apisix-nginx-module-${apisix_nginx_module_ver}
 fi
 
 if [ "$repo" == wasm-nginx-module ]; then
     cp -r "$prev_workdir" ./wasm-nginx-module-${wasm_nginx_module_ver}
 else
-    git clone --depth=1 -b $wasm_nginx_module_ver \
-        https://github.com/api7/wasm-nginx-module.git \
+    git clone --depth=1 -b fix_header_files \
+        https://github.com/zll600/wasm-nginx-module.git \
```
Comment (monkeyDluffy6017) [inline build-apisix-base.sh:70]: please revert this

File: build-apisix-base.sh [hunk 6]
```diff
@@ -42,8 +42,8 @@ tar -zxvpf openresty-${OPENRESTY_VERSION}.tar.gz > /dev/null
 if [ "$repo" == ngx_multi_upstream_module ]; then
     cp -r "$prev_workdir" ./ngx_multi_upstream_module-${ngx_multi_upstream_module_ver}
 else
-    git clone --depth=1 -b $ngx_multi_upstream_module_ver \
-        https://github.com/api7/ngx_multi_upstream_module.git \
+    git clone --depth=1 -b upgrade_openresty-1.25.3 \
+        https://github.com/zll600/ngx_multi_upstream_module.git \
```
Comment (monkeyDluffy6017) [inline build-apisix-base.sh:46]: please revert this

File: build-apisix-runtime.sh (hunk 3 above)
Comment (zll600) [reply build-apisix-runtime.sh:95]: Should we need to add a new tag for `ngx_multi_upstream_module` first?

File: build-apisix-runtime.sh [hunk 7]
```diff
@@ -131,8 +131,8 @@ fi
 if [ "$repo" == grpc-client-nginx-module ]; then
     cp -r "$prev_workdir" ./grpc-client-nginx-module-${grpc_client_nginx_module_ver}
 else
-    git clone --depth=1 -b $grpc_client_nginx_module_ver \
-        https://github.com/api7/grpc-client-nginx-module \
+    git clone --depth=1 -b fix_header_files \
+        https://github.com/zll600/grpc-client-nginx-module \
```
[the remark this answers is not in the document]
Comment (zll600) [reply build-apisix-runtime.sh:135]: Should we need to add a new tag for `grpc-client-nginx-module` first?

File: build-apisix-runtime.sh [hunk 8]
```diff
@@ -107,16 +107,16 @@ fi
...
         apisix-nginx-module-${apisix_nginx_module_ver}
 fi
 
 if [ "$repo" == wasm-nginx-module ]; then
     cp -r "$prev_workdir" ./wasm-nginx-module-${wasm_nginx_module_ver}
 else
-    git clone --depth=1 -b $wasm_nginx_module_ver \
-        https://github.com/api7/wasm-nginx-module.git \
+    git clone --depth=1 -b fix_header_files \
+        https://github.com/zll600/wasm-nginx-module.git \
```
Comment (zll600) [inline build-apisix-runtime.sh:119]: Should we need to add a new tag for `wasm-nginx-module` first?

File: build-apisix-runtime.sh [hunk 9]
```diff
@@ -158,6 +158,12 @@ grpc_engine_path="-DNGX_GRPC_CLI_ENGINE_PATH=$OR_PREFIX/libgrpc_engine.so -DNGX_
 cd openresty-${OPENRESTY_VERSION} || exit 1
 
 or_limit_ver=0.08
+enable_http3=${enable_http3:-}
```
Comment (membphis) [inline build-apisix-runtime.sh:161]: why do we need to support `enable_http3` as an input flag?

enable http3 as default, I think it is acceptable

Comment (membphis) [reply build-apisix-runtime.sh:161]: as we discussed by online meeting, I think we can remove `if ... fi`

the next APISIX will contain http3 feature

**Comment by membphis** [comment]
pls resolve your conflict, you need to rebase your branch

**Review by membphis — APPROVED** [review]

**Pull request merged**